optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintCollector.java

```diff
@@ -53,7 +53,7 @@
 
     /**
      * A lambda that converts the result container into the result.
-     * @return never null
+     * @return null when the result would be invalid, such as maximum value from an empty container.
```

Comment:
Good point. Now I understand.
The user never calls this method - and the user can never run into an NPE with it due to the groupBy() way of using it, but this javadoc is indeed correct and it's an important fact.